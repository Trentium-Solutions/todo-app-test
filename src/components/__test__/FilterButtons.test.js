import React from 'react'
import {render, screen} from '@testing-library/react'
import {createRoot} from 'react-dom/client'
import {filters} from '../../functions/changeNote'
import FilterButtons from '../FilterButtons'
import TestRenderer from 'react-test-renderer'

describe('filterbuttons test suite', () => {
  const filterNames = Object.keys(filters)
  const filterList = filterNames.map((name) => (
    <FilterButtons key={name} name={name} />
  ))
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const root = createRoot(div)
    root.render(filterList)
  })
  it('renders buttons correctly', () => {
    render(filterList)
    expect(screen.getAllByTestId('filterbuttons')).toBeTruthy()
  })

  it('matches snapshot buttons', () => {
    const tree = TestRenderer.create(filterList).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
