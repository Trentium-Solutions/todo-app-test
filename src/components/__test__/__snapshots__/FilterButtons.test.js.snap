// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterbuttons test suite matches snapshot buttons 1`] = `
Array [
  <button
    className="btn btn-secondary rounded p-2 btn-sm mx-2 my-3 w-25 small mybtn"
    data-testid="filterbuttons"
    onClick={[Function]}
    type="button"
  >
    All
  </button>,
  <button
    className="btn btn-secondary rounded p-2 btn-sm mx-2 my-3 w-25 small mybtn"
    data-testid="filterbuttons"
    onClick={[Function]}
    type="button"
  >
    Active
  </button>,
  <button
    className="btn btn-secondary rounded p-2 btn-sm mx-2 my-3 w-25 small mybtn"
    data-testid="filterbuttons"
    onClick={[Function]}
    type="button"
  >
    Completed
  </button>,
  <button
    className="btn btn-secondary rounded p-2 btn-sm mx-2 my-3 w-25 small mybtn"
    data-testid="filterbuttons"
    onClick={[Function]}
    type="button"
  >
    Deleted
  </button>,
]
`;
